File: dio-codes/src/intermediate/heranca_bancaria/HerancaBancaria.kt
package intermediate.heranca_bancaria

import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.OutputStreamWriter

private fun getInput(): Array<CharSequence> {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val input: Array<CharSequence> = arrayOf(reader.readLine(), reader.readLine(), reader.readLine(), reader.readLine())
    reader.close()
    return input
}

private open class BankAccount(
    open val owner: CharSequence, open val accountNumber: CharSequence, open val balance: CharSequence
)

private class SavingsAccount(
    override val owner: CharSequence,
    override val accountNumber: CharSequence,
    override val balance: CharSequence,
    val interestRate: CharSequence
) : BankAccount(owner, accountNumber, balance)

fun main() {
    val input: Array<CharSequence> = getInput()
    val savingsAccount = SavingsAccount(input[0], input[1], input[2], input[3])
    val output: CharSequence =
        "Conta Poupanca:\r\n${savingsAccount.owner}\r\n${savingsAccount.accountNumber}\r\nSaldo: R\$ ${savingsAccount.balance}\r\nTaxa de juros: ${savingsAccount.interestRate}%"
    OutputStreamWriter(System.out).append(output).close()
}

File: dio-codes/src/intermediate/heranca_bancaria/herancaBancaria.ts
import reader from "readline-sync"

class BankAccount {
    constructor(readonly owner: string, readonly accountNumber: string, readonly balance: string) {
    }
}

class SavingsAccount extends BankAccount {
    constructor(owner: string, accountNumber: string, balance: string, readonly taxaJuros: string) {
        super(owner, accountNumber, balance)
    }
}

function main() {
    let input = [reader.question(), reader.question(), reader.question(), reader.question()]
    let contaPoupanca = new SavingsAccount(input[0], input[1], input[2], input[3])
    const output: string = `Conta Poupanca:\r\n${contaPoupanca.owner}\r\n${contaPoupanca.accountNumber}\r\nSaldo: R$ ${contaPoupanca.balance}\r\nTaxa de juros: ${contaPoupanca.taxaJuros}%`
    console.log(output)
}

main()